server/src-lib/Hasura/GraphQL/Schema/Action.hs

```diff
@@ -1,380 +1,293 @@
 module Hasura.GraphQL.Schema.Action
-  ( mkActionsSchema
+  ( actionExecute
+  , actionAsyncMutation
+  , actionAsyncQuery
   ) where
 
-import qualified Data.HashMap.Strict           as Map
-import qualified Language.GraphQL.Draft.Syntax as G
-
-import           Hasura.GraphQL.Schema.Builder
-import           Hasura.GraphQL.Schema.Common  (mkDescriptionWith)
-
-import           Hasura.GraphQL.Resolve.Types
-import           Hasura.GraphQL.Validate.Types
 import           Hasura.Prelude
+import           Data.Has
+
+import qualified Data.Aeson                            as J
+import qualified Data.HashMap.Strict                   as Map
+import qualified Language.GraphQL.Draft.Syntax         as G
+
+import qualified Hasura.GraphQL.Parser                 as P
+import qualified Hasura.GraphQL.Parser.Internal.Parser as P
+import qualified Hasura.RQL.DML.Internal               as RQL
+import qualified Hasura.RQL.DML.Select.Types           as RQL
+
+import           Hasura.GraphQL.Parser                 (FieldParser, InputFieldsParser, Kind (..),
+                                                        Parser, UnpreparedValue (..))
+import           Hasura.GraphQL.Parser.Class
+import           Hasura.GraphQL.Schema.Common
+import           Hasura.GraphQL.Schema.Select
 import           Hasura.RQL.Types
 import           Hasura.Session
 import           Hasura.SQL.Types
+import           Hasura.SQL.Value
 
-mkAsyncActionSelectionType :: ActionName -> G.NamedType
-mkAsyncActionSelectionType = G.NamedType . unActionName
-
-mkAsyncActionQueryResponseObj
-  :: ActionName
-  -- Name of the action
-  -> GraphQLType
-  -- output type
-  -> ObjTyInfo
-mkAsyncActionQueryResponseObj actionName outputType =
-  mkHsraObjTyInfo
-  (Just description)
-  (mkAsyncActionSelectionType actionName) -- "(action_name)"
-  mempty -- no arguments
-  (mapFromL _fiName fieldDefinitions)
-  where
-    description = G.Description $ "fields of action: " <>> actionName
-
-    mkFieldDefinition (fieldName, fieldDescription, fieldType) =
-      mkHsraObjFldInfo
-      (Just fieldDescription)
-      fieldName
-      mempty
-      fieldType
-
-    fieldDefinitions = map mkFieldDefinition
-      [ ( "id", "the unique id of an action"
-        , G.toGT $ mkScalarTy PGUUID)
-      , ( "created_at", "the time at which this action was created"
-        , G.toGT $ mkScalarTy PGTimeStampTZ)
-      , ( "errors", "errors related to the invocation"
-        , G.toGT $ mkScalarTy PGJSON)
-      , ( "output", "the output fields of this action"
-        , unGraphQLType outputType)
-      ]
-
-mkQueryActionField
-  :: ActionName
+actionExecute
```

Comment:
`actionExecute` didn't seem very intuitive to me at the first glance, can we rename this function to `actionSyncMutation` (like we have `actionAsyncMutation`) or something else?

Reply:
Oh, Now I understand why it's called `actionExecute` after going through `query'` in `Hasura.GraphQL.Schema`. No need to change anything here 🙂

Reply:
@codingkarthik Can you please explain the name? If you write it here I'd be happy to commit it.

Reply:
I'll document the function with a haddock comment and commit it 😄

Reply:
@codingkarthik This is resolved, right?

Reply:
Oh sorry, I forgot to resolve it. Doing it now!